src/store/models/designer.ts

```diff
@@ -213,6 +231,101 @@ const designerModel: DesignerModel = {
       }
     });
   }),
+  renameLightningNode: thunk(
+    async (actions, { nodeId, name, backendName }, { getState, getStoreActions }) => {
+      const { allCharts, activeId } = getState();
+      const chart = allCharts[activeId];
+
+      // Ensure the node exists
+      const node = chart.nodes[nodeId];
+      if (!node) {
+        throw new Error(`Lightning node with id ${nodeId} not found.`);
+      }
+
+      // Update the node chart
+      const updatedChart = {
+        ...chart,
+        nodes: {
+          ...chart.nodes,
+          [name]: {
+            ...chart.nodes[nodeId],
+            id: name,
+          },
+        },
+      };
+
+      // Update the state with the modified chart
+      allCharts[activeId] = updatedChart;
+      // remove the old node with previous key
+      getStoreActions().designer.removeNode(nodeId);
+      getStoreActions().designer.updateBackendLink({
+        lnName: name,
+        backendName: backendName,
+      });
+    },
+  ),
+  renameBitcoinNode: thunk(
```

Comment:
I don't think it's necessary to have 3 dirrent rename node functions. Their implementations looks pretty much identical. I would try to condense these into one.